### ISSUE TITLE
Payload names inconsistency

### ISSUE THREAD

**Issue by ShellCode33** [rapid7/metasploit-framework#13788]
Hello there, I'm coding a tool around msfvenom, and I noticed Windows and GNU/Linux 32 bits payloads do not follow the same naming convention :

32 bits payloads :
```
windows/meterpreter/reverse_https
linux/x86/meterpreter/reverse_https
```

64 bits payloads :
```
windows/x64/meterpreter/reverse_https
linux/x64/meterpreter/reverse_https
```

It's not that big of an issue, but it forces me to handle special cases instead of having something as simple as :
```python
subprocess.check_output(["msfvenom", "--payload", f"{platform}/{arch}/meterpreter/reverse_https", [...]
```

Moreover some people might think `windows/meterpreter/reverse_https` is a 64bits payload, so adding `x86` to it might help !

So I was thinking maybe someone could do something about it ? I might have missed something somewhere, feel free to close the issue :smiley:

**Comment by okoibraun** [comment]
looking at this, my thoughts are, its just a simple creation "x86"
directory in the 32-bit version of the windows-meterpreter directory by
rapid7 and moving the meterpreter directory into the newly created "x86"
directory and shipping the new release, so it can be universally available
for all users.

**Comment by bcoles** [comment]
> its just a simple creation "x86"

Yeah backwards compatibility is for suckers

**Comment by okoibraun** [comment]
Agreed. for us and @ShellCode, we can simply write a workaround, around it.
but you really need to think about the suckers out there (just as you've
said).
If @ShellCode, noticed it, what bout others, who haven't?

**Comment by bcoles** [comment]
It is likely expected that Windows x86 payloads will run on Windows x64 operating systems, hence defaulting to x86 payloads.

However, a consistent naming format would be nice. It would make more sense to create an alias for `windows/x86/*` payloads, rather than moving the entire directory and breaking everything that depends on it.

**Comment by okoibraun** [comment]
Yeah, true that. i forgot about the dependency(ies) part. I guess, we
all/@ShellCode just need to work around it (the [arch] thing). 👍👍👍

**Comment by egypt** [comment]
+1 for aliasing. This is a frequent source of confusion among new users, but The Olds have muscle memory and scripts that rely on the arch-less names.
